### PR TITLE
feat: switch codemirror to prism

### PR DESCRIPTION
**Pull request by amritk**
Should be lighter, definitely has more control, lets see. 

Comes with json, bash, js, and css pre-loaded. Then the rest of the languages are auto loaded on the fly

@cameronrohani we should probably ~~make our own css file which ingests the theme vars~~ exposed some css classes so we can theme the code as well

### CODE REVIEW

**Comment by hanspagel** [comment]
Great work! I’d be tempted to rename the component to `CodeBlock`. In HTML, Markdown and rich text `code` refers to inline code. Wdyt?

**Comment by amritk** [comment]
Sounds like a good plan. Or even code highlight or something

File: packages/api-reference/src/components/CodeBlock/CodeBlock.vue [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+})
+</script>
+<template>
+  <pre
+    :class="{
+      'line-numbers': lineNumbers,
+    }"><code ref="el" :class="`lang-${language}`">{{content}}</code></pre>
+</template>
+<style>
+code[class*='language-'],
```
Comment (amritk) [inline packages/api-reference/src/components/CodeBlock/CodeBlock.vue:59]: This was just copy/pasted from one of their themes

File: packages/api-reference/src/components/CodeBlock/CodeBlock.vue [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  },
+)
+
+const el = ref(null)
+const language = computed(() => {
+  return props.lang === 'node' ? 'js' : props.lang
+})
+
+plugins.autoloader.languages_path =
+  'https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/'
```
Comment (marclave) [inline packages/api-reference/src/components/CodeBlock/CodeBlock.vue:39]: i know we are loading these async but should we have this bundled with the repo? :)

Comment (amritk) [reply packages/api-reference/src/components/CodeBlock/CodeBlock.vue:39]: I always prefer cdn over bundle for better performance! But doesn't matter too much these days

Comment (marclave) [reply packages/api-reference/src/components/CodeBlock/CodeBlock.vue:39]: alright im down, we might need to bundle it for some of our integrations but this is cool : )

**Review by marclave — COMMENTED** [review]
nice work @amritk ! ✨

**Review by marclave — APPROVED** [review]

**Pull request merged**